src/main/java/core/basesyntax/service/impl/TransactionProcessor.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.TransactionParser;
+import java.util.List;
+
+public class TransactionProcessor {
+    private final OperationStrategy operationStrategy;
+    private final TransactionParser transactionParser;
+
+    public TransactionProcessor(OperationStrategy operationStrategy,
+                                TransactionParser transactionParser) {
+        this.operationStrategy = operationStrategy;
+        this.transactionParser = transactionParser;
+    }
+
+    public void processTransactions(List<String> inputData) {
+        inputData.stream()
+                .map(transactionParser::parseLine)
+                .forEach(transaction -> {
```

Comment:
The `processTransactions` method seems to be doing too much. It's parsing the input data and applying the operations. Consider splitting these responsibilities into separate methods to adhere to the Single Responsibility Principle (SRP) of the SOLID rules.

Reply:
The logic was refactored